### ISSUE TITLE
FloatLabel: Add support for SelectOneMenu,AutoComplete,DatePicker

### ISSUE THREAD

**Issue by willbigas** [primefaces/primefaces#7011]
**Description**

Could we have floatlabel in SelectOneMenu, AutoComplete, DatePicker like we have in PrimeReact, today I do some css workarounds to have something like that, it would be very interesting to have this in primefaces for jsf.
https://primefaces.org/primereact/showcase/#/floatlabel

**Describe the solution you would like**
Using the same strategy that we have in traditional inputs, have the possibility of placing the float-label behavior in the inputs (SelectOneMenu, AutoComplete, DatePicker).

**Additional context**
Float label is very interesting, when there are many inputs inside a form consuming less space on the grid, besides that it is much more modern than traditional labels, due to its js/css animation.

**Example XHTML**
```xml
              <span class="ui-float-label">
                <p:selectOneMenu id="select" value="#{selectOneMenuView.selectedOption}">
                    <f:selectItem itemLabel="Option1" itemValue="Option1"/>
                    <f:selectItem itemLabel="Option2" itemValue="Option2"/>
                    <f:selectItem itemLabel="Option3" itemValue="Option3"/>
                   </p:selectOneMenu>
                <p:outputLabel for="select" value="Float Label here..."/>
              </span>
```

**Comment by melloware** [comment]
I can look and see how PrimeReact is doing it and maybe modify ours.  Our current float label was based on PrimeReact.

**Comment by melloware** [comment]
I think this one is for @mertsincan as looking at PrimeReact they do some CSS in the core and the rest in themes.

**Comment by mertsincan** [comment]
This is slightly different from PrimeReact. The structure of PrimeFaces is not very suitable for this and some tricks are needed. Please check ultima-jsf example;
https://www.primefaces.org/ultima/floatlabel.xhtml
Is it Ok for you?

floatlabel.xhtml;
```
<div class="card" style="width: 100%">
            <h5>Float Label</h5>
            <p>All input text components support floating labels by adding (
                <mark>.ui-float-label</mark>
                ) to wrapper class.
            </p>
            <div class="p-grid ui-fluid p-mt-3 floatlabel-demo">
                <div class="p-col-12 p-md-6">
                    <div class="p-field">
                        <span class="ui-float-label">
                            <p:inputText/>
                            <p:outputLabel for="@previous" value="InputText"/>
                        </span>
                    </div>
                    <div class="p-field">
                        <span class="ui-float-label ui-input-icon-left">
                            <i class="pi pi-search"/>
                            <p:inputText/>
                            <p:outputLabel for="@previous" value="Left Icon" styleClass="p-ml-4"/>
                        </span>
                    </div>
                    <div class="p-field">
                        <span class="ui-float-label ui-input-icon-right">
                            <p:inputText/>
                            <p:outputLabel for="@previous" value="Right Icon"/>
                            <i class="pi pi-spin pi-spinner"/>
                        </span>
                    </div>
                    <div class="p-field">
                        <span class="ui-float-label">
                            <p:inputMask mask="99/99/9999"/>
                            <p:outputLabel for="@previous" value="InputMask"/>
                        </span>
                    </div>
                    <div class="p-field">
                        <span class="ui-float-label">
                            <p:inputNumber/>
                            <p:outputLabel for="@previous" value="InputNumber"/>
                        </span>
                    </div>
                    <div class="p-field">
                        <span class="ui-float-label">
                            <p:chips/>
                            <p:outputLabel for="@previous" value="Chips"/>
                        </span>
                    </div>
                </div>

                <div class="p-col-12 p-md-6">
                    <div class="p-field">
                        <div class="ui-inputgroup">
                            <span class="ui-inputgroup-addon">
                                <i class="pi pi-user"/>
                            </span>
                            <span class="ui-float-label" style="width: 100%">
                                <p:inputText/>
                                <p:outputLabel for="@previous" value="InputGroup"/>
                            </span>
                        </div>
                    </div>
                    <div class="p-field">
                        <span class="ui-float-label">
                            <p:autoComplete/>
                            <p:outputLabel for="@previous" value="AutoComplete"/>
                        </span>
                    </div>
                    <div class="p-field">
                        <span class="ui-float-label">
                            <p:datePicker/>
                            <p:outputLabel for="@previous" value="Datepicker"/>
                        </span>
                    </div>
                    <div class="p-field">
                        <span class="ui-float-label">
                            <p:selectOneMenu id="cities" autoWidth="false">
                                <f:selectItem itemLabel="" itemValue="" />
                                <f:selectItem itemLabel="New York" itemValue="New York" />
                                <f:selectItem itemLabel="Rome" itemValue="Rome" />
                                <f:selectItem itemLabel="London" itemValue="London" />
                                <f:selectItem itemLabel="Istanbul" itemValue="Istanbul" />
                                <f:selectItem itemLabel="Paris" itemValue="Paris" />
                            </p:selectOneMenu>
                            <p:outputLabel for="@previous" value="Dropdown" />
                        </span>
                    </div>
                    <div class="p-field">
                        <span class="ui-float-label">
                            <p:inputTextarea rows="3" cols="30"/>
                            <p:outputLabel for="@previous" value="Textarea"/>
                        </span>
                    </div>
                </div>
            </div>
        </div>
```

**Comment by melloware** [comment]
@willbigas can you try Merts suggeston above and see if it meets your needs?

**Comment by willbigas** [comment]
@melloware , I believe that this way should work, thanks!

**Comment by melloware** [comment]
I have submitted a PR and created a showcase example rolling all Mert's Ultima customizations into the core PF.

**Comment by Teutron-flo** [comment]
@melloware, one short hint here, I think I found a bug in your code.
I added your changes in a local primfaces 11 build and tested my application with it.

For selectOneMenu, it works when you focus the component and also when you change values.
But in case the component has a value and is disabled, nothing happen:

![image](https://user-images.githubusercontent.com/80260448/171798262-d435598e-c219-4314-a603-dfbe6bce4c96.png)
